pliers/filters/text.py

```diff
@@ -60,14 +65,36 @@ def __init__(self, stemmer='porter', tokenize=True, *args, **kwargs):
                              "instance of class StemmerI.")
         self.stemmer = stemmer
         self.tokenize = tokenize
+        self.case_sensitive = case_sensitive
         super(WordStemmingFilter, self).__init__()
 
     def _filter(self, stim):
+
+        pos_map = {
+            'ADJ': 'a',
+            'ADJ_SAT': 's',
+            'ADV': 'r',
+            'NOUN': 'n',
+            'VERB': 'v'
+        }
+
+        def pos_wordnet(txt):
+            pos_tagged = dict(nltk.pos_tag(txt, tagset='universal'))
+            pos_tagged = {tok: pos_map[tag] if tag in pos_map else 'n'
+                          for tok, tag in pos_tagged.items()}
+            return pos_tagged
+
+        tokens = [stim.text]
         if self.tokenize:
-            tokens = stim.text.split()
+            tokens = nltk.word_tokenize(tokens[0])
+            tokens = [tok if self.case_sensitive else tok.lower()
```

Comment:
Maybe de-dent this line so it's outside the conditional? Not sure if lower-casing has any impact in the non-tokenized case, but if it does (or just in case), it should be applied in both cases.

Reply:
fixed in latest commit